rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/api/Coordinator.java

```diff
@@ -377,7 +377,7 @@ public Response compensateNestedLRA(@PathParam("NestedLraId") String nestedLraId
     @PUT
     @Path("nested/{NestedLraId}/forget")
     public Response forgetNestedLRA(@PathParam("NestedLraId") String nestedLraId) {
-        lraService.remove(toURI(nestedLraId));
+        lraService.remove(lraService.getTransaction(toURI(nestedLraId)));
```

Comment:
This code calls into LRAService to convert an id into a transaction and then calls back into lraService with the returned transaction. Why not add a method to LRAService that takes the id (and internally converts it):

```java
public boolean remove(URI lra)
```

Reply:
I think the `LRAService#remove` should work directly with `lraId` and not with `Transaction`.The `remove` method should probably use the LRA `Transaction` instance obtained from the list and not the one that comes in argument
- then this line is currently wrong: https://github.com/jbosstm/narayana/blob/5.10.5.Final/rts/lra/lra-coordinator-jar/src/main/java/io/narayana/lra/coordinator/domain/service/LRAService.java#L207

Reply:
the issue is in the other call of the `remove` method where the recovered transaction object on which we actually call the second phase again and has its state updated accordingly (failedToClose) is not included in the lra list. The LRA taken from the list is in the wrong state.

Probably the better fix would be to include the Recovery LRA object in the lra list directly but that would need to be replaced on every recovery pass as a new recovery LRA is created and so far we didn't run into this issue so I've decided that this is an easier fix.

Let me know how would you like to proceed, please.

Reply:
@xstefank for me to understand better, the code that we discuss is
```
if (lra.isFailed()) { // persist failed LRA state
  lra.deactivate();
}
```
where the `lra` is deactivated and this call has to be run on a `lra` instance, right?
You suggest that the passed `lra` is not part of the `Coordinator` `lras` when `remove` is called from `LRAService#finished`, correct?
If the `lra` is not included in the lra list (which I assume is meant to be `lras`) then the code I pasted above won't never be called. Then it won't be a problem. 

Probably. So I understand it in a wrong way. Can you explain me a bit more, please?

Reply:
Just keep the existing  `lraService.remove` method and maybe add a new one that takes a transaction instance. I would advice against refactoring the code at this stage.

Reply:
I understand the issue now and I got the point about the `lra` instance passed to the `remove` method. 
I was trying to understand the processing and I'm not able to come with a better/nicer way. I assume it's fine to be this way.

I don't like code duplication thus I do not prefer the ducplication code into `lraService.remove` method but if it's fine for the rest then ok.